custom-docker/build.sh

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+usage() {
+    cat <<EOF 1>&2
+Usage: $0 [-h] [-a <ARCHITECTURES>] [-r <VERSION>]
+Builds the Trino Docker image
+
+-h       Display help
+-a       Build the specified comma-separated architectures, defaults to amd64,arm64,ppc64le
+-t       Image tag name, defaults to trino-custom
+-r       Build the specified Trino release version, downloads all required artifacts
+EOF
+}
+
+ARCHITECTURES=(amd64 arm64 ppc64le)
+# Set the desires Trino version. This version is also used automatically in the
+# Dockerfile.
+TRINO_VERSION='474'
+echo "Trino version set to ${TRINO_VERSION}"
+TAG_PREFIX=trino-custom
+
+# Download a plugin from any URL. Must be a Trino plugin zip file.
+# Specify the URL and the name for the desired folder in the plugin directory.
+function downloadUrlPlugin() {
+    url=$1
+    folder=$2
+    filename=$(basename ${url})
+    echo "Download ${url}"
+    curl ${url} --fail --location --silent --output ${filename}
+    unzip -q ${filename}
+    tmp=$(basename ${filename} .zip)
+    echo "Add plugin ${tmp} as ${folder} directory"
+    mv ${tmp} customization/usr/lib/trino/plugin/${folder}
+    rm ${filename}
+}
+
+# Download a plugin from Maven Central.
+# Specify the group path (groupId translated to a path), artifactid, version,
+# and name for the desired folder in the plugin directory.
+function downloadPlugin() {
+    groupPath=$1
+    artifactid=$2
+    version=$3
+    folder=$4
+    filename="${artifactid}-${version}.zip"
+    url="https://repo1.maven.org/maven2/${groupPath}/${artifactid}/${version}/$filename"
+    downloadUrlPlugin ${url} ${folder}
+}
+
+# Download any Trino plugin from Maven Central.
+# Specify the group path (groupId translated to a path), artifactid, version, 
+# and name for the desired folder in the plugin directory.
+function downloadGavPlugin() {
+    groupPath=$1
+    artifactid=$2
+    version=$3
+    folder=$4
+    downloadPlugin ${groupPath} ${artifactid} ${version} ${folder}
+}
+
+# Download a core Trino plugin with path io/trino from Maven Central.
+# Specify artifactid, version, and name for the desired folder in the plugin directory.
+function downloadTrinoPlugin() {
+    artifactid=$1
+    version=$2
+    folder=$3
+    downloadGavPlugin 'io/trino' ${artifactid} ${version} ${folder}
+}
+
+# Download the desired Trino plugins.
+# Use the functions above as desired to download plugins from Maven Central
+# or other URLs.
+function processPluginsForCustomizations() {
+    echo "Download plugins for customization"
+    downloadTrinoPlugin 'trino-ai-functions' ${TRINO_VERSION} 'ai-functions'
+    downloadTrinoPlugin 'trino-blackhole' ${TRINO_VERSION} 'blackhole'
+    downloadTrinoPlugin 'trino-faker' ${TRINO_VERSION} 'faker'
+    downloadTrinoPlugin 'trino-jmx' ${TRINO_VERSION} 'jmx'
+    downloadTrinoPlugin 'trino-memory' ${TRINO_VERSION} 'memory'
+    #downloadTrinoPlugin 'trino-tpch' ${TRINO_VERSION} 'tpch'
```

Comment:
There are catalogs using these plugins, so the container will fail to start.

Reply:
Yeah .. but next line has equivalent call as example..